components/designSystem/NetworkSelector/Dropdown.tsx

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import React, {useState} from 'react';
+import {DropdownMenuCheckboxItem, DropdownMenuContent, DropdownMenuSeparator} from 'components/Primitives/DropdownMenu';
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
+import {supportedNetworks} from '@utils/tools.chains';
+import {IconChevronBottom} from '@yearn-finance/web-lib/icons/IconChevronBottom';
+import {cl} from '@yearn-finance/web-lib/utils/cl';
+
+import type {InputHTMLAttributes, ReactElement} from 'react';
+import type {TNDict} from '@yearn-finance/web-lib/types';
+
+export function NetworkDropdownSelector(
+	props: {
+		value: number[];
+		onChange: (value: number[]) => void;
+	} & Omit<InputHTMLAttributes<HTMLInputElement>, 'value' | 'onChange'>
+): ReactElement {
+	/**************************************************************************
+	 * If some networks are passed as props, we want to use them. As we are
+	 * using a TNDict to store the selected networks, we need to convert the
+	 * array of networks into a TNDict.
+	 *************************************************************************/
+	const [selectedNetworks, set_selectedNetworks] = useState<TNDict<boolean>>(
+		props.value.reduce((acc, networkId) => {
+			acc[networkId] = true;
+			return acc;
+		}, {} as TNDict<boolean>)
```

Comment:
It should be possible to get rid of type casting here by doing this:
`(acc: TNDict<boolean>, networkId) => {}`